src/app/mapControls/Filter.js

```diff
@@ -0,0 +1,227 @@
+        },
+        itemClicked: function (value) {
+            // summary:
+            //      description
+            // value: String
+            //      value of the item that was clicked
+            console.log('app/mapControls/Filter:itemClicked', arguments);
+
+            var index = this.selectedValues.indexOf(value);
+            if (index === -1) {
+                this.selectedValues.push(value);
+            } else {
+                this.selectedValues.splice(index, 1);
+            }
+
+            this.numSpan.innerHTML = this.selectedValues.length;
+
+            this.emit('changed');
+        },
+        getQuery: function () {
+            // summary:
+            //      assembles all selected values into a def query
+            console.log('app/mapControls/Filter:getQuery', arguments);
+
+            if (this.selectedValues.length) {
+                var values;
+                if (this.fieldType === c.TYPE_TEXT) {
+                    values = this.selectedValues.map(function (v) {
+                        return "'" + v + "'";
+                    });
+                } else {
+                    values = this.selectedValues;
+                }
+                if (this.any) {
+                    var where = this.fieldName + ' IN (' + values.join(', ') + ')';
+                    return this.getRelatedTableQuery(where);
+                } else {
+                    var that = this;
+                    return values.reduce(function (previousReturn, currentValue) {
+                        var where = that.fieldName + ' = ' + currentValue;
+                        if (!previousReturn) {
+                            return that.getRelatedTableQuery(where);
+                        } else {
+                            return previousReturn + ' AND ' + that.getRelatedTableQuery(where);
+                        }
+                    }, false);
+                }
+            } else {
+                return undefined;
+            }
+        },
+        toggleAny: function () {
+            // summary:
+            //      description
+            console.log('app/mapControls/Filter:toggleAny', arguments);
+
+            var that = this;
+            setTimeout(function () {
+                that.any = domClass.contains(that.anyBtn, 'active');
+                that.emit('changed');
+            }, 0);
+        },
+        remove: function () {
+            // summary:
+            //      removes widget from visible filters
+            console.log('app/mapControls/Filter:remove', arguments);
+
+            this.clear();
+
+            this.emit('removed', this);
+        },
+        open: function () {
+            // summary:
+            //      opens the body of the filter
+            console.log('app/mapControls/Filter:open', arguments);
+
+            $(this.body).collapse('show');
+
+            this.inherited(arguments);
+        },
+        getRelatedTableQuery: function (where) {
```

Comment:
If I select a few items from the feature type list and I have `all` selected i get a sql string like 

```sql
PROJECT_ID IN(SELECT Project_ID FROM POINT WHERE TypeCode=0) OR 
Project_ID IN(SELECT Project_ID FROM LINE WHERE TypeCode=0) OR 
Project_ID IN(SELECT Project_ID FROM POLY WHERE TypeCode=0) AND 
Project_ID IN(SELECT Project_ID FROM POINT WHERE TypeCode=1) OR 
Project_ID IN(SELECT Project_ID FROM LINE WHERE TypeCode=1) OR 
Project_ID IN(SELECT Project_ID FROM POLY WHERE TypeCode=1) AND 
Project_ID IN(SELECT Project_ID FROM POINT WHERE TypeCode=3) OR 
Project_ID IN(SELECT Project_ID FROM LINE WHERE TypeCode=3) OR 
Project_ID IN(SELECT Project_ID FROM POLY WHERE TypeCode=3) AND 
Project_ID IN(SELECT Project_ID FROM POINT WHERE TypeCode=7) OR 
Project_ID IN(SELECT Project_ID FROM LINE WHERE TypeCode=7) OR 
Project_ID IN(SELECT Project_ID FROM POLY WHERE TypeCode=7) AND 
Project_ID IN(SELECT Project_ID FROM POINT WHERE TypeCode=8) OR 
Project_ID IN(SELECT Project_ID FROM LINE WHERE TypeCode=8) OR 
Project_ID IN(SELECT Project_ID FROM POLY WHERE TypeCode=8)
```
which I think is producing incorrect results.

Will you update the query to 

```sql
Project_ID IN(
	SELECT Project_ID FROM POINT WHERE TypeCode IN(0,1,3,7,8)
	intersect 
	SELECT Project_ID FROM LINE WHERE TypeCode IN(0,1,3,7,8)
	intersect
	SELECT Project_ID FROM POLY WHERE TypeCode IN(0,1,3,7,8)
)
```
for `any` it can be simplified to 

```sql
Project_ID IN(
	SELECT Project_ID FROM POINT WHERE TypeCode IN(0,1,3,7,8)
	union 
	SELECT Project_ID FROM LINE WHERE TypeCode IN(0,1,3,7,8)
	union
	SELECT Project_ID FROM POLY WHERE TypeCode IN(0,1,3,7,8)
)
```